include/openthread/platform/logging.h

```diff
@@ -161,6 +161,16 @@ typedef enum otLogRegion
  */
 void otPlatLog(otLogLevel aLogLevel, otLogRegion aLogRegion, const char *aFormat, ...);
 
+/**
+ * This function handles OpenThread log level changes.
+ *
+ * This platform function is called whenever the OpenThread log level changes.
```

Comment:
Should we add in the docs that this is optional platform API (weake one used if not provided by platform layer)?

And maybe that it is applicable only when we have `OPENTHREAD_CONFIG_LOG_LEVEL_DYNAMIC_ENABLE`?

Reply:
Will do.